File: Javascript/String/Goal-Parser-Interpretation.js
/**
 * @param {string} command
 * @return {string}
 */
const interpret = (command) => {
  let result = "";

  for (let i = 0; i < command.length; i++) {
    let char = command[i];

    if (char == "G") result += "G";

    if (char == "(") result += command[i + 1] == ")" ? "o" : "al";
  }

  return result;
};

/**
 * @param {string} command
 * @return {string}
 */
const interpret = (command) => {
  return command.replaceAll("()", "o").replaceAll("(al)", "al");
};

File: Rust/Goal-Parser-Interpretation.rs
pub struct Solution {}

impl Solution {
    pub fn interpret(command: String) -> String {
        let mut result = String::new();
        let mut i = 0;

        let chars: Vec<char> = command.chars().collect();

        while i < command.len() {
            if chars[i] == 'G' {
                result.push_str("G");
            }

            if chars[i] == '(' {
                let x = if chars[i + 1] == ')' { "o" } else { "al" };

                result.push_str(x);
            }
            i += 1;
        }

        result
    }

    pub fn interpret_replace(command: String) -> String {
        command.replace("()", "o").replace("(al)", "al")
    }
}

fn main() {}
